modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/rel/LimitNode.java

```diff
@@ -136,6 +153,6 @@ protected Downstream<RowT> requestDownstream(int idx) {
 
     /** {@code True} if requested 0 results, or all already processed. */
     private boolean fetchNone() {
-        return (fetchNode != null && fetch == 0) || (fetch > 0 && rowsProcessed == fetch + offset);
+        return (!fetchUndefined && fetch == 0) || (fetch > 0 && rowsProcessed == fetch + offset);
```

Comment:
I'd rename the method to smth like `hasMoreData()`

Reply:
I`d prefer to stay it as is, we will change it in scope of execution refactoring

Reply:
```suggestion
        return fetchUndefined || rowsProcessed < fetch + offset;
```

Reply:
done